src/core/ip-address-access.c

```diff
@@ -188,3 +188,26 @@ IPAddressAccessItem* ip_address_access_reduce(IPAddressAccessItem *first) {
 
         return first;
 }
+
+bool ip_address_access_item_is_any(IPAddressAccessItem *first) {
+        bool ipv4_any = false, ipv6_any = false;
+        IPAddressAccessItem *a;
+        int i = 0;
+
+        LIST_FOREACH(items, a, first) {
+                if (i > 1)
+                        return false;
+
+                if (a->prefixlen == 0) {
+                        if (a->family == AF_INET)
+                                ipv4_any = in4_addr_is_null(&a->address.in);
```

Comment:
similar here, if `in4_addr_is_null(&a->address.in)` doesn't hold, we can immediately return false, no?